app/scripts/modules/core/src/pipeline/filter/executionFilter.service.ts

```diff
@@ -368,4 +371,49 @@ export class ExecutionFilterService {
     ExecutionState.filterModel.asFilterModel.clearFilters();
     ExecutionState.filterModel.asFilterModel.applyParamsToUrl();
   }
+
+  public static getRelation() {
+    return flatten(this.getRelations());
+  }
+
+  public static getRelations(): any {
+    const relation = SETTINGS.filterRelations;
+    const relationKeys = Object.keys(relation);
+    const relationVal = Object.values(relation);
+    const relationArr: any = [];
+    if (relationKeys.length) {
+      relationVal.forEach((value, index) => {
+        if (value === null) {
+          let key = index;
+          const rel = [];
+          while (key !== -1) {
+            rel.push(relationKeys[key]);
+            key = relationVal.indexOf(relationKeys[key]);
+          }
+          relationArr.push(rel);
+        }
+      });
+      return relationArr;
```

Comment:
Is the type of the `relationArr` an array of `string[]`? If so, I would change the type from `any`.